src/usr/local/buildpack/utils/cache.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# will attempt to download the file at the given url and stores it in the cache.
+# If this file already exists, will return the cached version
+# The cache will only used if BUILDPACK_CACHE_DIR is set
+# First argument is the url, second one is the filename (optional)
+function get_from_url () {
+  local url=${1}
+  check url true
+
+  local name
+  name=${2:-$(basename "${url}")}
+
+  if [ -n "${BUILDPACK_CACHE_DIR}" ] && [ -e "${BUILDPACK_CACHE_DIR}/${name}" ]; then
```

Comment:
We need to make sure the files are unique, can we use `TOOL_NAME` and `TOOL_VERSION` to create some implicit folders?

Reply:
I would like not to rely on the tool name and the tool version in this functions. 
Even if we agree that this is currently the only use case we should find a different solution.

The filename is not unique enough for you, so what about if we add the hash of the url to the name?
I would expect the full url to be unique enough without introducing the dependency on the tool name and version.

Reply:
hash would be ok for me, as filename isn't unique enough.